atc/worker/client.go

```diff
@@ -278,6 +278,58 @@ func (client *client) RunTaskStep(
 
 	select {
 	case <-ctx.Done():
+		// if we have an interrupt timeout, allow for a graceful termination
+		interruptTimeout, ok := InterruptTimeoutFromContext(ctx)
```

Comment:
Could you use the 
`if timeout,ok := InterruptTimeoutFromContext(ctx); ok {` 
style with an `else` to make the `else` case more explicit